src/reduce.test.js

```diff
@@ -11,9 +11,41 @@
     delete Array.prototype.reduce2;
   });
 
-  it('should ', () => {
+  it(`should return error if callback doesn't passed`, () => {
+    expect(() => {
+      [1, 2, 3].reduce2();
+    }).toThrow();
+  });
+
+  it('should work with undefined passed as an initial value', () => {
+    const callback = (a, b) => a + b;
+
+    expect([1, 2, 3].reduce2(callback, undefined)).toBe(NaN);
+    expect(['1', '2', '3'].reduce2(callback, undefined)).toBe('undefined123');
```

Comment:
The expectation `toBe('undefined123')` might not be correct. When `undefined` is used as an initial value, the result of the reduction might not be as expected due to JavaScript's type coercion. Verify the logic in the `reduce` implementation to ensure it handles undefined initial values correctly.